tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/NewLiferayWorkspaceOpMethods.java

```diff
@@ -63,21 +72,93 @@ public static final Status execute( final NewLiferayWorkspaceOp op, final Progre
+                final String modulesDir = op.getModulesDir().content( true );
+                final String themesDir = op.getThemesDir().content( true );
+                final String warsDir = op.getWarsDir().content( true );
+
+                ElementList<AdditionalProperty> additionalProperties = op.getAdditionalProperties();
+
+                Properties properties = new Properties();
+                File gradleProperties = wsPath.append( "gradle.properties" ).toFile();
+
+                try(InputStream io = new FileInputStream( gradleProperties );
+                                OutputStream out = new FileOutputStream( gradleProperties ))
+                {
+                    properties.load( io );
+                    properties.put( "liferay.workspace.environment", env );
+                    if( !env.equals( "common" ) || !env.equals( "dev" ) || !env.equals( "local" ) ||
+                        !env.equals( "prod" ) || !env.equals( "uat" ) )
+                    {
+                        File envFile = wsPath.append( "configs" ).append( env ).toFile();
+                        envFile.mkdirs();
+                        File portalExtProperties = new File( envFile, "portal-ext.properties" );
+                        portalExtProperties.createNewFile();
+                    }
+
+                    properties.put(
+                        "liferay.workspace.modules.default.repository.enabled", String.valueOf( repoEnabled ) );
+
+                    properties.put( "liferay.workspace.modules.dir", modulesDir );
+                    if( !modulesDir.equals( "modules" ) )
+                    {
+                        wsPath.append( "modules" ).toFile().delete();
+                        wsPath.append( modulesDir ).toFile().mkdirs();
+                    }
+
+                    properties.put( "liferay.workspace.themes.dir", themesDir );
+                    if( !themesDir.equals( "themes" ) )
+                    {
+                        wsPath.append( "themes" ).toFile().delete();
+                        wsPath.append( themesDir ).toFile().mkdirs();
+                    }
+
+                    properties.put( "liferay.workspace.wars.dir", warsDir );
+                    if( !warsDir.equals( "wars" ) )
+                    {
+                        wsPath.append( "wars" ).toFile().delete();
+                        wsPath.append( warsDir ).toFile().mkdirs();
+                    }
+
+                    for( AdditionalProperty p : additionalProperties )
+                    {
+                        String value = p.getValue().content( true );
+
+                        if( value == null )
+                        {
+                            value = "";
+                        }
+
+                        properties.put( p.getName().content( true ), value );
+                    }
+
+                    properties.store( out, null );
```

Comment:
We are overwritting exsiting properties file automatically.  The properties file by default may have documentation in the future.  we need to leave existing file as is and only write new property values if they have changed from the defaults.  Otherwise users wont know the defaults.

Use the library we have in liferay.core for writing properties files without overwriting existing ones "PropertiesConfiguration" or something like that.  

If they user picks all defaults, then we shouldn't touch the gradle.properties at all.